src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,230 @@
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tableRefObject,
+  visibleColumnsLength,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns || (stickyColumns.first === 0 && stickyColumns.last === 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  // Compute left table padding
+  const table = tableRefObject.current;
+  const tableLeftPadding = table ? Number(getComputedStyle(table).paddingLeft.slice(0, -2)) : 0;
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lasRightStickyColumnIndex = visibleColumnsLength - 1 - last + (hasSelection ? 1 : 0);
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets?.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets?.last[lasRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tableLeftPadding < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [
+    containerWidth,
+    noStickyColumns,
+    stickyColumns,
+    totalStickySpace,
+    visibleColumnsLength,
+    tableLeftPadding,
+    isWrapperScrollable,
+  ]);
+
+  useEffect(() => {
+    // Create new refs for the visible columns and selection column, if present
+    setTableCellRefs(tableCellRefs =>
+      [...new Array(visibleColumnsLength + (hasSelection ? 1 : 0))].map(
+        (_: any, i: number) => tableCellRefs[i] || createRef<HTMLTableCellElement>()
+      )
+    );
+  }, [visibleColumnsLength, hasSelection]);
+
+  const getStickyStyles = useCallback(
+    (colIndex: number, isStickyLeft: boolean, isStickyRight: boolean) => {
+      // Determine which side to apply sticky styles to
+      const stickySide = isStickyLeft ? 'left' : isStickyRight ? 'right' : '';
```

Comment:
if `isStickyLeft` and `isStickyRight` can't be both set to true or false, would it make sense for the getStickyStyles function to accept `stickySide` instead?